regolith/helpers/l_abstracthelper.py

```diff
@@ -103,18 +102,15 @@ def sout(self):
                           and rc.loc_inst.casefold() in presentation.get('location',"")]
             filtered_inst = [presentation for presentation in presentations
                              if presentation.get('type') in SEMINAR_TYPES and
-                             rc.loc_inst.casefold() in presentation.get('institution',"").casefold()
-                             and presentation not in exceptions]
+                             rc.loc_inst.casefold() in presentation.get('institution',"").casefold()]
             filtered_loc = [presentation for presentation in presentations
-                            if rc.loc_inst.casefold() in presentation.get('location',"").casefold()
-                            and presentation not in exceptions]
+                            if rc.loc_inst.casefold() in presentation.get('location',"").casefold()]
             if exceptions:
-                warn("These presentations have %s in both 'location' and 'institution'" % (rc.loc_inst))
-            for presentation in exceptions:
-                print("---------------------------------------")
-                print(f"Title: {presentation.get('title')}")
-                print(f"location: {presentation.get('location')}")
-                print(f"institution: {presentation.get('institution')}")
+                for presentation in exceptions:
+                    raise Exception("This presentation has %s in both 'location' and 'institution'" % (rc.loc_inst),
```

Comment:
- [ ] use an `fstring`
- [ ] raise ValueError
- [ ] tell the user the offending presentation, not just "This presentation" if you can
- [ ] move it up tp the top so it takes less time to crash if it is going to crash
- [ ] it is ok for this to be here (this is a lister), but it is more useful making sure the adder/updaters don't allow it in the first place.  They may not, but now is a great time to check while we are thinking about it.
- [ ] it needs a test

One more thought.  If we are even more clever we can do this check in the database validation.  google for cerberus and "conditional validation" or sthg like that.  Then we don't need any of htis code, which is kind of ugly anyway....

That is much more elegant a solution so worth looking into then we can just remove all this.

Reply:
Should I implement the cerberus solution in l_abstracthelper.py or in schemas.py?

Reply:
schemas.py.  It may also be necessary to tweak `regolith.validate`